File: mirea/contests/2026-04-10-MIREA-Spring-Madness-Contest-2026/D.cpp
// математика, жадные алгоритмы, O(n)
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
ll solve(int n, ll k, const vector<int> &a) {
    ll sum = 0, mx = 0;
    for (int i = 0; i < n; ++i) {
        ll x = a[i];
        sum += x;
        mx = max(mx, x);
    }
    ll answ, others = sum - mx;
    if (mx <= others + 1)
        answ = k * sum - (k - 1) * 1LL * n;
    else
        answ = k * (2 * others + 1) - (k - 1) * 1LL * (n - 1);
    return answ;
}
main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int t; cin >> t;
    while (t --> 0) {
        int n; ll k;
        cin >> n >> k;
        vector<int> a(n);
        for (int i = 0; i < n; i++)
            cin >> a[i];
        cout << solve(n, k, a) << '\n';
    }
}

File: mirea/contests/2026-04-10-MIREA-Spring-Madness-Contest-2026/D.py
# математика, жадные алгоритмы, O(n)
import sys

input = sys.stdin.readline

def solve(n, k, a):
    sum_ = 0
    mx = 0
    for i in range(n):
        x = a[i]
        sum_ += x
        mx = max(mx, x)
    others = sum_ - mx
    if mx <= others + 1:
        answ = k * sum_ - (k - 1) * n
    else:
        answ = k * (2 * others + 1) - (k - 1) * (n - 1)
    return answ

t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    a = list(map(int, input().split()))
    print(solve(n, k, a))